### ISSUE TITLE
Search gets stuck

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#18077]
See https://github.com/osmandapp/OsmAnd/issues/17290#issuecomment-1703757083. Looks like this is a separate issue, independent of recent track fixes, as I can also reproduce on older builds.

Only  on older builds it seems the search 'recovers' earlier, after like 15 seconds.

PS: Visiting the Configure map > Tracks screen causes the same effect.

PS2: The issue is created even if immediately after tapping MyPlaces > Tracks you tap the (top left) back button (i.e. even before the spinner indicating the screen is being prepared stops spinning).

PS3: The issue is NOT caused when going to MyPlaces / Favorites tab, only after (even if just briefly) visiting the MyPlaces/Tracks tab or Configure Map/Tracks screen.

**Comment by DmitryAlexei** [comment]
Tried with OsmAnd~ 4.6.0#710m, released: 2023-09-01 and OsmAnd 4.6.0, released: 2023-09-01:
- Fresh App start
- Search > Categories > Accommodations. The first page of search hits almost instantaneously.
- Go back to the map, then go to the My Places > Tracks screen.
- Go back to the map screen, repeated the search for accommodations second time. Same result.
 
Every time, search results are shown almost immediately. 
Note that I only have 100 tracks, so for me the track screen loads very quickly.

**Comment by sonora** [comment]
Perhaps this is somehow memory handling related? Could it be that the processes of opening the Tracks screen and using Search somehow affect each other via blocking memory the other proces would need, or similar?

PS: I had temporarily renamed my tracks folder (70 subfolders, about 3500 tracks total) for one app start, so OsmAnd reset the gpx database. Ever since, opening the Tracks screen now "only" takes about 5-7 seconds the first time round (before it was up to 30), 3-6 on subsequent attempts. (This aeemes to indicate that sonehow over time the gpx database can deteriorate and affect performance, but this is likely a separate issue.)

**Comment by sonora** [comment]
Has nobody been able to reproduce this yet? Every single one of our 4.6 beta builds suffers the effect, making any version after 4.5.10 useless for me. Unless I am the only one affected, this may constitute quite an issue for our 4.6 release.

**Comment by vshcherb** [comment]
Search of tracks ? configure map? or POI? issue name not very specific. It might be related to filters activity.

**Comment by sonora** [comment]
No, I am talking about the Search button on the map screen, performing e.g. a Category search (or a search for a town), see detailed steps as linked sbove.

I just notice that when this condition of a 'stuck search' had happened, the app seems constatly busy with some search core code: Even after I had aborted the search and left the search scrern, then went to the Dev plugin Settings to view the log file, the log screen is busy permanently scrolling screens like this without interruption, thousands of lines. Only restarting the app stops this:

![Screenshot_20230915-205212_OsmAnd~](https://github.com/osmandapp/OsmAnd/assets/845735/c4977de2-16ce-468f-85e3-5ef9de6d8c24)

**Comment by sonora** [comment]
I could meanwhile establish that the issue is closely connected with having a significant number of tracks in OsmAnd's track folder. The issue is not present, or causes only smaller search delays of 1-10 sec, if you have not very many tracks.

**Comment by sonora** [comment]
More information regarding

> The issue is not present, or causes only smaller search delays of 1-10 sec, if you have not very many tracks.

Here is another surprising impact of having many tracks on the device and search screen performance:

Fresh start of OsmAnd, then inmediately tap the Search button on the map screen: If you have comparatively few tracks stored in OsmAnd's track folder (e.g. 200), the search screen pops up almost instantaneously (and displays the search histor). Under the same corcumstances, if I have all my 35000 tracks in the folder (even if none of them is visible on the map), it takes some 30 or more seconds until the search history displays on the searhj screen (and even if the history is empty or even disabled) and the search screen finally becomes responsive.

**Comment by vshcherb** [comment]
Same as https://github.com/osmandapp/OsmAnd/issues/18355

**Comment by Chumva** [comment]
@sonora can you test nightly after [this fix](https://github.com/osmandapp/OsmAnd/pull/18482)

**Comment by sonora** [comment]
Unfortunately, that fix had no visible effect on this issue here.

The issue gets created immediately (and it seems lastingly until the next app sraer), and 100% reproducible on my setup, even if I just **briefly** enter the My Tracks screen, then immediately use the top left back button (even not waiting the x seconds it always takes for the screen to show the track folders).

**Comment by Chumva** [comment]
Can you add logcat logs from new nightly build with this issue?

**Comment by sonora** [comment]
@Chumva Vitaliy, I have to revise my comment above: After playing around with the new build a few more minutes now and restarting the app several times, it seems the issue has indeed now disappeared! Not sure why it was still present initially, perhaps some cache or database was slow in picking it up! I will watch the situation, but in the last 5 minutes it was consistently gone. **Good job!** 😉
